compiler/rustc_interface/src/util.rs

```diff
@@ -192,7 +192,16 @@ pub(crate) fn run_in_thread_pool_with_globals<F: FnOnce(CurrentGcx) -> R + Send,
             // `TyCtxt` TLS reference here.
             let query_map = current_gcx2.access(|gcx| {
                 tls::enter_context(&tls::ImplicitCtxt::new(gcx), || {
-                    tls::with(|tcx| QueryCtxt::new(tcx).collect_active_jobs())
+                    tls::with(|tcx| {
+                        let (query_map, complete) = QueryCtxt::new(tcx).collect_active_jobs();
+                        if !complete {
+                            eprintln!("internal compiler error: failed to get query map in deadlock handler, aborting process");
+                            // We need to abort here as we failed to resolve the deadlock,
+                            // otherwise the compiler could just hang,
```

Comment:
Why does failing to `collect_active_jobs` mean that we failed to resolve the deadlock?
Why the compiler can hang in this case?

Reply:
Deadlock detection depends on obtaining unfinished query tasks in the current state and collecting them into the `QueryMap`.

Reply:
> Deadlock detection depends on obtaining all the unfinished query tasks in the state and collecting them into one `QueryMap`.

@Zoxc Could you add this as a comment to code (preferably with a brief explanation of why it depends on obtaining all the tasks).